pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentUploadDownloadRestletResource.java

```diff
@@ -1007,6 +1008,8 @@ public Response endReplaceSegments(
       @ApiParam(value = "OFFLINE|REALTIME", required = true) @QueryParam("type") String tableTypeStr,
       @ApiParam(value = "Segment lineage entry id returned by startReplaceSegments API", required = true)
       @QueryParam("segmentLineageEntryId") String segmentLineageEntryId,
+      @ApiParam(value = "Trigger an immediate segment prune") @QueryParam("prune") @DefaultValue("false")
```

Comment:
I wouldn't call it `prune`. I feel `cleanup` might be a better name

Reply:
I tried to avoid `cleanup` because it was already used on some other places and didn't want to mix concepts but after further review yeah, I think is harmless to do so.